tests/tracking/test_fluent.py

```diff
@@ -681,3 +683,98 @@ def test_delete_tag():
     with pytest.raises(MlflowException):
         mlflow.delete_tag("b")
     mlflow.end_run()
+
+
+@pytest.fixture
+def library_to_mlflow_module():
+    return {
+        "tensorflow": "tensorflow",
+        "keras": "keras",
+        "fastai": "fastai",
+        "pyspark": "spark",
+        "sklearn": "sklearn",
+        "xgboost": "xgboost",
+        "lightgbm": "lightgbm",
+        "mxnet.gluon": "gluon",
+    }
+
+
+def clear_imports_and_import_hooks():
+    import wrapt
+
+    for integration_name in library_to_mlflow_module.keys():
+        try:
+            del sys.modules[integration_name]
+        except Exception:
+            pass
+        try:
+            del wrapt.importer._post_import_hooks[integration_name]
+        except Exception:
+            pass
+
+
+@pytest.mark.large
+def test_universal_autolog_does_not_throw_if_specific_autolog_throws(
+    mocker, library_to_mlflow_module,
+):
+    clear_imports_and_import_hooks()
+
+    for integration_name in library_to_mlflow_module.keys():
+        with mock.patch(
+            "mlflow." + library_to_mlflow_module[integration_name] + ".autolog"
+        ) as autolog_mock:
+            autolog_mock.side_effect = Exception("asdf")
+            mlflow.autolog()
+            importlib.__import__(integration_name)
+
+    clear_imports_and_import_hooks()
+
+
+@pytest.mark.large
+def test_universal_autolog_calls_specific_autologs_correctly(mocker, library_to_mlflow_module):
```

Comment:
Ditto here - can we parametrize by `library_to_mlflow_module`? This will run a separate test case for each library / MLflow module pair.